source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -449,6 +450,11 @@ EnvoyQuicServerStream::validateHeader(absl::string_view header_name,
   return result;
 }
 
+void EnvoyQuicServerStream::OnMetadataComplete(size_t /*frame_len*/,
+                                               const quic::QuicHeaderList& header_list) {
+  request_decoder_->decodeMetadata(metadataMapFromHeaderList(header_list));
```

Comment:
do we need to do any validation steps with or without UHV? I honestly don't recall sorry

Reply:
Ok I didn't know if we had concerns about character validation, since I think Envoy has some assumptions there, but maybe not for metadata.
cc @yanavlasov

Reply:
@birenroy might also know

Reply:
No restrictions on metadata character set, since it is opt-in and the strings are length prefixed.